typescript/src/ethereum.ts

```diff
@@ -562,3 +571,86 @@ class WalletRegistry extends EthereumContract {
     return publicKey.substring(2)
   }
 }
+
+/**
+ * Implementation of the Ethereum TBTCVault handle.
+ */
+export class TBTCVault extends EthereumContract implements ChainTBTCVault {
```

Comment:
I believe we should refactor the file structure. It's hard to navigate through code if we keep all implementations in one file. I think we could create a dir for each chain and then create dirs for each handle. For example:
```
└── ethereum/
    ├── index.ts // Export all from bridge, vault etc.
    ├── bridge/
    │   └── index.ts // Ethereum implementation of the Bridge handle
    ├── contract/
    │   └── index.ts // Common impl of the Ethereum contract
    └── vault/
        └── index.ts // Ethereum implementation of the Vault handle
```

Ofc we can address it in a separate PR just flagging.

Reply:
Yeah, we need refactoring!

Reply:
We should think about refactoring but I'm not sure if the proposed file structure with multiple nestings will be easier to follow than one file with several hundreds of lines of code. I think we can group files but aim for a more flat and simpler structure:
```
ethereum
    bridge.ts
    vault.ts
    index.ts
```

Reply:
@nkuba what about this one? Do we plan to address it in this PR or separately? In the case of the latter, can be leave a TODO?

Reply:
1edcb3960228d1313fea8341b28c8f773a8ae21b

Reply:
Ok, leaving this discussion unresolved for future reference.